File: backend/src/db/seed.js
import crypto from 'crypto';
import db from './schema.js';

const seed = () => {
  db.exec('DELETE FROM route_incidents; DELETE FROM route_lighting;');

  const latBase = 28.6139;
  const lngBase = 77.2090;
  const types = ['mugging', 'harassment', 'poor_visibility', 'theft', 'stalking'];
  const severities = ['low', 'medium', 'high'];
  const times = ['morning', 'afternoon', 'evening', 'night', 'late_night'];

  for (let i = 0; i < 20; i++) {
    db.prepare('INSERT INTO route_incidents (id, lat, lng, type, severity, time_of_day, description) VALUES (?, ?, ?, ?, ?, ?, ?)').run(
      crypto.randomUUID(),
      latBase + (Math.random() - 0.5) * 0.02,
      lngBase + (Math.random() - 0.5) * 0.02,
      types[Math.floor(Math.random() * types.length)],
      severities[Math.floor(Math.random() * severities.length)],
      times[Math.floor(Math.random() * times.length)],
      'Mock incident'
    );
  }

  for (let i = 0; i < 15; i++) {
    db.prepare('INSERT INTO route_lighting (id, lat, lng, coverage_score, time_of_day) VALUES (?, ?, ?, ?, ?)').run(
      crypto.randomUUID(),
      latBase + (Math.random() - 0.5) * 0.02,
      lngBase + (Math.random() - 0.5) * 0.02,
      Math.random() * 0.8 + 0.1,
      times[Math.floor(Math.random() * times.length)]
    );
  }

  console.log('Seeded 20 incidents and 15 lighting points');
  console.log('Seeded data pending live data partner');
};

seed();

File: backend_fastapi/app/db/seed.py
import uuid
from app.db.mongodb import get_db

def seed_db():
    db = get_db()
    
    # Clear existing
    incidents_col = db["route_incidents"]
    lighting_col = db["route_lighting"]
    
    incidents = [
        {"_id": str(uuid.uuid4()), "lat": 28.6140, "lng": 77.2095, "type": "mugging", "severity": "high", "time_of_day": "night", "description": "Attempted robbery near park line"},
        {"_id": str(uuid.uuid4()), "lat": 28.6145, "lng": 77.2098, "type": "poor_visibility", "severity": "medium", "time_of_day": "night", "description": "Broken street lights for 200m"},
        {"_id": str(uuid.uuid4()), "lat": 28.6150, "lng": 77.2105, "type": "harassment", "severity": "high", "time_of_day": "late_night", "description": "Reported stalking incident"},
        {"_id": str(uuid.uuid4()), "lat": 28.6160, "lng": 77.2120, "type": "theft", "severity": "low", "time_of_day": "evening", "description": "Pickpocketing near market exit"}
    ]
    
    lighting = [
        {"_id": str(uuid.uuid4()), "lat": 28.6139, "lng": 77.2090, "coverage_score": 0.9, "time_of_day": "night"},
        {"_id": str(uuid.uuid4()), "lat": 28.6140, "lng": 77.2095, "coverage_score": 0.2, "time_of_day": "night"},
        {"_id": str(uuid.uuid4()), "lat": 28.6150, "lng": 77.2110, "coverage_score": 0.8, "time_of_day": "night"},
        {"_id": str(uuid.uuid4()), "lat": 28.6160, "lng": 77.2120, "coverage_score": 0.4, "time_of_day": "night"}
    ]
    
    for inc in incidents:
        incidents_col.insert_one(inc)
    for lgt in lighting:
        lighting_col.insert_one(lgt)
        
    print(f"[INFO] Seeded MongoDB with {len(incidents)} incidents and {len(lighting)} lighting data points.")

if __name__ == "__main__":
    seed_db()
